### PR TITLE
create one testcase per error on JUnit format

### PR DESCRIPTION
**Pull request by linuxmaniac**
Same file as mentioned on #27 but with this change
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="Luacheck report" tests="22">
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:1" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W111" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:21:1: setting non-standard global variable 'NGCPAvp'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:2" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W111" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:24:1: setting non-standard global variable 'NGCPAvp_MT'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:3" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:25:15: accessing undefined variable 'NGCPAvp'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:4" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W112" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:28:14: mutating non-standard global variable 'NGCPAvp'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:5" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W212" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:28:21: unused argument 'self'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:6" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W112" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:30:9: mutating non-standard global variable 'NGCPAvp_MT'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:7" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W431" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:30:38: shadowing upvalue 't' on line 29"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:8" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:34:24: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:9" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W211" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:36:23: unused variable 'i'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:10" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W211" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:36:26: unused variable 'v'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:11" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W421" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:37:21: shadowing definition of variable 'i' on line 36"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:12" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:41:17: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:13" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:43:17: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:14" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:49:20: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:15" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W112" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:51:9: mutating non-standard global variable 'NGCPAvp_MT'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:16" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W431" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:51:42: shadowing upvalue 't' on line 29"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:17" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:52:27: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:18" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:55:33: accessing undefined variable 'NGCPAvp_MT'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:19" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W112" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:58:14: mutating non-standard global variable 'NGCPAvp'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:20" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:62:9: accessing undefined variable 'sr'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:21" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W112" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:65:14: mutating non-standard global variable 'NGCPAvp'"/>
    </testcase>
    <testcase name="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:22" classname="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua">
        <failure type="W113" message="/home/vseva/Sipwise/git/lua-ngcp-kamailio/ngcp/avp.lua:69:9: accessing undefined variable 'sr'"/>
    </testcase>
</testsuite>
```
Not sure about the need of the num_test/num_tests

### CODE REVIEW

File: src/luacheck/format.lua [hunk 1]
```diff
@@ -211,20 +211,38 @@ end
 
 function formatters.JUnit(report, file_names)
    local buf = {[[<?xml version="1.0" encoding="UTF-8"?>]]}
+   local num_tests = 0
+   local num_test = 0
 
-   table.insert(buf, ([[<testsuite name="Luacheck report" tests="%d">]]):format(#report))
+   for _, file_report in ipairs(report) do
+      if file_report.error or file_report ~= 0 then
```
Comment (mpeterv) [inline src/luacheck/format.lua]: `file_report ~= 0` must be `#file_report ~= 0`?

**Comment by mpeterv** [comment]
Is it necessary to put index of each testcase into its name? I think it'd be better to only do that for failures, where name is not unique otherwise, and reset the count for each file.

**Comment by linuxmaniac** [comment]
> Is it necessary to put index of each testcase into its name?
I would say that name should be unique for a testsuite. Any other idea?

> I think it'd be better to only do that for failures, where name is not unique otherwise, and reset the count for each file.

Sure, you're right.I will push another version tomorrow.

Thanks!

**Comment by mpeterv** [comment]
@linuxmaniac I've just pushed my version (#29), can you try it?

**Comment by linuxmaniac** [comment]
implemented in a better way at #29

**Pull request closed without merging**